executor/prune.go

```diff
@@ -0,0 +1,169 @@
+package executor
+
+import (
+	"errors"
+	"strconv"
+	"strings"
+
+	k8upv1alpha1 "github.com/vshn/k8up/api/v1alpha1"
+	"github.com/vshn/k8up/constants"
+	"github.com/vshn/k8up/job"
+	"github.com/vshn/k8up/observer"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// PruneExecutor will execute the batch.job for Prunes.
+type PruneExecutor struct {
+	generic
+	prune *k8upv1alpha1.Prune
+}
+
+// NewPruneExecutor will return a new executor for Prune jobs.
+func NewPruneExecutor(config job.Config) *PruneExecutor {
+	return &PruneExecutor{
+		generic: generic{config},
+	}
+}
+
+// Execute creates the actual batch.job on the k8s api.
+func (p *PruneExecutor) Execute() error {
+	pruneObject, ok := p.Obj.(*k8upv1alpha1.Prune)
+	if !ok {
+		return errors.New("object is not a prune")
+	}
+
+	p.prune = pruneObject
+
+	if pruneObject.GetK8upStatus().Started {
+		return nil
+	}
+
+	jobObj, err := job.GetGenericJob(pruneObject, p.Config)
+	jobObj.GetLabels()[job.K8upExclusive] = "true"
+	if err != nil {
+		return err
+	}
+
+	// FIXME(mw): I just copied this over. I'm unsure if that's really necessary.
+	go func() {
+		p.startPrune(jobObj)
```

Comment:
having this in a separate goroutine: I'm not sure if this is needed and not sure if it's a good idea.

Possible issue: `p.prune = pruneObject` -- this could be a race condition. I copied parts from the backup executor but I'm not quite sure about why some things are done as they are done :)

Reply:
Yeah that part of the backup is a very special case because we have to wait for the pre-backup pods to be ready and don't want to slow down the queue execution.

It's not really needed here.

Reply:
As the go routine here isn't really require you can change `startPrune()` to return an error.